DragaliaAPI.Test/Controllers/QuestControllerTest.cs

```diff
@@ -112,4 +115,116 @@ public async Task ReadStory_ProducesExpectedResponse()
         this.mockStoryService.VerifyAll();
         this.mockUpdateDataService.VerifyAll();
     }
+
+    [Fact]
+    public async Task OpenTreasure_ProducesExpectedResponse()
```

Comment:
I'd prefer a test in the integration test project that tests the whole thing rather than just the controller if possible